File: Backend/ChatApp.Domain/Entities/Account.cs
using ChatApp.Domain.Enums;

namespace ChatApp.Domain.Entities
{
    public class Account
    {
        public long UserID { get; set; }
        public string Username { get; set; }
        public AccountType AccountType { get; set; }
        public DateTime CreatedTime { get; set; }
        public bool IsActive { get; set; }
        public bool IsOnline { get; set; }
        public DateTime LastSeen { get; set; }
    }
}

File: Frontend/src/types/account.ts
import { AccountType } from "./enums";
import { IUserProfile } from "./profile";

export interface IAccount {
  UserID: number;
  Username: string;
  AccountType: AccountType;
  CreatedTime: Date;
  IsActive: boolean;
  IsOnline: boolean;
  LastSeen: Date;
}

export interface ILoginResponse {
  accessToken: string;
  refreshToken: string;
  userProfile: IUserProfile;
}

export interface ILoginRequest {
  username: string;
  password: string;
}

export interface IRegisterRequest {
  firstName: string;
  lastName: string;
  username: string;
  password: string;
}

export interface IRegisterResponse {
  accessToken: string;
  refreshToken: string;
  userProfile: IUserProfile;
}
